### PR TITLE
HPCC-16011 Add a daliadmin dfscheck command to locate dfs problems

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
jira: https://track.hpccsystems.com/browse/HPCC-16011

**Comment by ghalliday** [comment]
@jakesmith please review

File: dali/daliadmin/daliadmin.cpp [hunk 1]
```diff
@@ -658,6 +659,43 @@ static void writeGroup(IGroup *group, const char *name, const char *outputFilena
...
+    }
+    path.setLength(prevLength);
+    return issues;
+}
+
+void dfsCheck()
+{
+    StringBuffer xpath;
+    Owned<IRemoteConnection> conn = connectXPathOrFile("/Files",true,xpath);
+    if (!conn) {
```
Comment (jakesmith) [inline dali/daliadmin/daliadmin.cpp]: style: newline

File: dali/daliadmin/daliadmin.cpp [hunk 2]
```diff
@@ -658,6 +659,43 @@ static void writeGroup(IGroup *group, const char *name, const char *outputFilena
...
+        issues += dfsCheck(path, &elems->query());
+    }
+    path.setLength(prevLength);
+    return issues;
+}
+
+void dfsCheck()
+{
+    StringBuffer xpath;
+    Owned<IRemoteConnection> conn = connectXPathOrFile("/Files",true,xpath);
```
Comment (jakesmith) [inline dali/daliadmin/daliadmin.cpp]: minor: as never a File here, think clearer if uses connect() directly.

**Comment by jakesmith** [comment]
Minor comments.

This is (currently) a very specific check for a bug.
I wonder if it wouldn't have been better to extend listmatches or similar to have made it easier to track down the context of the reported error.

@ghalliday

**Comment by ghalliday** [comment]
The idea for adding a new command was that it was much easier for a naive user to use.

Whenever we found a problem we should extend the code so that it finds instances of the new problem.

**Comment by ghalliday** [comment]
@jakesmith please check again

**Comment by jakesmith** [comment]
@ghalliday - Looks good

**Comment by ghalliday** [comment]
@richardkchapman

**Pull request merged**